setup/ansible-inventory

```diff
@@ -33,14 +33,6 @@ test-requireio_davglass-ubuntu1404-arm_odroid_xu-2
 test-digitalocean-ubuntu12-x64-1
 test-digitalocean-ubuntu12-x64-2
 
-[iojs-build-smartos]
```

Comment:
@geek The inventory now lives in [`ansible/inventory.yml`](https://github.com/nodejs/build/blob/master/ansible/inventory.yml#L108-L115), since we are transitioning to https://github.com/nodejs/build/blob/master/ansible/playbooks/jenkins/worker/create.yml

Reply:
Everything in `setup/` is obsolete and should be deleted once the `ansible/` equivalent is working.

Thanks @maclover7 for doing this work!